test/gtest/ucp/test_ucp_ep_reconfig.cc

```diff
@@ -520,3 +520,92 @@ UCS_TEST_P(test_reconfig_asymmetric, resolve_remote_id)
 }
 
 UCP_INSTANTIATE_TEST_CASE_TLS(test_reconfig_asymmetric, shm_ib, "shm,ib");
+
+class test_reconfigure_update_rank : public test_reconfig_asymmetric {
+public:
+    void init() override
+    {
+        test_ucp_ep_reconfig::init();
+        modify_config("NUM_EPS", std::to_string(num_eps()));
+    }
+
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_with_value(variants, UCP_FEATURE_TAG, 200, "mid");
+        add_variant_with_value(variants, UCP_FEATURE_TAG, 500, "large");
+    }
+
+    void create_entities_and_connect() override
+    {
+        create_entity(true, false);
+        create_entity(false, false);
+        sender().connect(&receiver(), get_ep_params());
+        receiver().connect(&sender(), get_ep_params());
+    }
+
+    unsigned num_eps() const
+    {
+        return get_variant_value();
+    }
+
+    void verify_configuration();
+};
+
+void test_reconfigure_update_rank::verify_configuration()
+{
+    auto r_sender   = static_cast<const entity*>(&sender());
+    auto r_receiver = static_cast<const entity*>(&receiver());
+
+    if (should_reconfigure()) {
+        EXPECT_TRUE(r_sender->is_reconfigured());
+        EXPECT_TRUE(r_receiver->is_reconfigured());
+    } else {
+        EXPECT_FALSE(r_sender->is_reconfigured());
+        EXPECT_FALSE(r_receiver->is_reconfigured());
+    }
+
+    r_sender->verify_configuration(*r_receiver, 0);
+    r_receiver->verify_configuration(*r_sender, 0);
+}
+
+UCS_TEST_P(test_reconfigure_update_rank, promote,
+           "DYNAMIC_TL_SWITCH_INTERVAL=1000s")

```

Comment:
why 1000s? is not it too long for test?

Reply:
it's intended to prevent `ucp_worker_track_ep_usage_always` from progressing the usage tracker

Reply:
also seems it is now the default value